dashboard/app/assets/javascripts/videos.js.erb

```diff
@@ -11,6 +11,19 @@ function createVideoWithFallback(parentElement, options, width, height) {
   return video;
 }
 
+function onVideoEnded() {
+  $('.video-modal').trigger("ended");
+}
+
+function onYouTubeIframeAPIReady() {
+  var player = YT.get('video') || new YT.Player('video');
+  player.addEventListener('onStateChange', function (state) {
+    if (state.data === YT.PlayerState.ENDED) {
+      onVideoEnded();
+    }
+  });
+}
+
```

Comment:
(i think that assumption should be ok to make within Code Studio currently, but we might want to make a note of that limitation somewhere in this code in case we end up changing that assumption in the future).